packages/lib-classifier/src/plugins/tasks/DrawingTask/models/DrawingAnnotation.spec.js

```diff
@@ -1,19 +1,187 @@
+        })
+        .create({
+          drawingTask,
+          annotation
+        })
+        snapshot = annotation.toSnapshot()
+      })
+
+      it('should contain exactly one annotation', function () {
+        expect(snapshot).to.have.lengthOf(1)
+      })
+
+      describe('the annotation snapshot', function () {
+        it('should contain a task key', function () {
+          const [ annotation ] = snapshot
+          expect(annotation.task).to.equal('T0')
+        })
+
+        it('should contain a taskType', function () {
+          const [ annotation ] = snapshot
+          expect(annotation.taskType).to.equal('drawing')
+        })
+
+        it('should include snapshots of the drawn marks', function () {
+          const [ annotation ] = snapshot
+          const pointSnapshot = {
+            details: [],
+            frame: 0,
+            toolIndex: 0,
+            toolType: 'point',
+            x: 50,
+            y: 100
+          }
+          expect(annotation.value).to.deep.equal([ pointSnapshot ])
+        })
+      })
+    })
+
+    describe('with subtasks', function () {
+      let snapshot
+
+      before(function () {
+        const drawingTask = DrawingTask.create({
+          instruction: 'draw things!',
+          taskKey: 'T0',
+          tools: [
+            {
+              type: 'point',
+              details: [{
+                type: 'single',
+                answers: ['yes', 'no'],
+                question: 'yes or no?'
+              }]
+            }
+          ],
+          type: 'drawing'
+        })
+        const annotation = DrawingAnnotation.create(drawingAnnotationSnapshot)
+        const mockStore = types.model('MockStore', {
+          drawingTask: DrawingTask,
+          annotation: DrawingAnnotation
+        })
+        .create({
+          drawingTask,
+          annotation
+        })
+        const pointTool = drawingTask.tools[0]
+        const point1 = pointTool.createMark({
+          x: 50,
+          y: 100
+        })
+        const point2 = pointTool.createMark({
+          x: 150,
+          y: 200
+        })
+        const questionTask = pointTool.tasks[0]
+        point1.addAnnotation(questionTask, 0)
+        point2.addAnnotation(questionTask, 1)
+        snapshot = annotation.toSnapshot()
+      })
+
+      it('should contain the task annotation plus one annotation for each subtask', function () {
+        expect(snapshot).to.have.lengthOf(3)
+      })
+      it('should contain snapshots of the annotation plus subtask answers', function () {
+        const [ annotation ] = snapshot
+        const point1Answer = {
+          task: 'T0.0.0',
+          taskType: 'single',
+          value: 0,
+          markIndex: 0
+        }
+        const point2Answer = {
+          task: 'T0.0.0',
+          taskType: 'single',
+          value: 1,
+          markIndex: 1
+        }
+        expect(snapshot).to.deep.equal([ annotation, point1Answer, point2Answer ])
```

Comment:
Here’s where we test that subtasks annotations are output from marks as expected.